tools/mpremote/mpremote/pyboardextended.py

```diff
@@ -211,15 +218,21 @@ def write(self, buf):
         c.wr_bytes(buf)
         n = c.rd_s32()
         c.end()
+        self.cur += n
         return n
 
-    def seek(self, n):
+    def seek(self, n, whence=SEEK_SET):
+        if whence == SEEK_END:
+            n += self.size
+        elif whence == SEEK_CUR:
+            n += self.cur
```

Comment:
Instead of handling this whence logic on the device side, it feels more robust to put it on the host/PC side.  Ie pass through whence as a new argument to CMD_SEEK.  Advantages of doing it that way:
1. less code on the device side (which is important for smaller devices)
2. semantics of whence are handled by calls to seek() on the host/PC side, therefore handled by the O/S and will match the mounted filesystem semantics
3. if the file changes size while open then SEEK_END will still work

Reply:
This makes perfect sense. I took a look and I seem to understand the commands a bit better now. I'll give it a try and swap it around! :)